File: app/src/main/java/com/example/hw8/PostDao.java
package com.example.hw8;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface PostDao {
    @Insert
    void insert(Post post);

    @Delete
    void delete(Post post);

    @Query("SELECT * FROM post")
    List<Post> getAllPosts();
}

File: app/src/main/java/com/example/hw8/PostDao.kt
package com.example.hw8

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface PostDao {
    @Insert
    suspend fun insert(post: Post)

    @Delete
    suspend fun delete(post: Post)

    @Query("SELECT * FROM post")
    suspend fun getAllPosts(): List<Post>
}